apps/explorer/src/pages/epochs/EpochDetail.tsx

```diff
@@ -36,6 +38,41 @@ function SuiStats({
     );
 }
 
+function getRefGasPrice(validators: SuiValidatorSummary[] | undefined) {
+    if (!validators?.length) {
+        return 0;
+    }
+
+    const REF_THRESHOLD = 66.67;
+
+    const sortedByGasPrice = [...validators].sort(
+        (a, b) => Number(a.gasPrice) - Number(b.gasPrice)
+    );
+    const totalStaked = validators.reduce(
+        (acc, cur) => acc + Number(cur.stakingPoolSuiBalance),
+        0
+    );
+
+    let sumOfStakes = 0;
+    let result = 0;
+
+    while (sumOfStakes <= REF_THRESHOLD) {
+        const validator = sortedByGasPrice.shift();
+        if (!validator) {
+            break;
+        }
+
+        const currentGasPrice = Number(validator.gasPrice);
+        const stake = Number(validator?.stakingPoolSuiBalance);
+        const stakeShare = (stake / totalStaked) * 100;
```

Comment:
All of these should be bigints

Reply:
Moved this code out into a util in `core` and updated to use `BigInt` conversions.